maestro-studio/server/src/main/java/maestro/studio/InsightService.kt

```diff
@@ -0,0 +1,35 @@
+package maestro.studio
+
+import com.fasterxml.jackson.annotation.JsonInclude
+import com.fasterxml.jackson.module.kotlin.jacksonObjectMapper
+import io.ktor.server.application.*
+import io.ktor.server.response.*
+import io.ktor.server.routing.*
+import kotlinx.coroutines.*
+import maestro.utils.Insight
+import maestro.utils.Insights
+import kotlin.coroutines.resume
+import kotlin.coroutines.suspendCoroutine
+
+object InsightService {
+
+    private var insight: Insight? = null
+
+    fun routes(routing: Route) {
+        routing.get("/api/banner-message") {
+            Insights.onInsightsUpdated {
+                insight = it.first { insight -> insight.visibility == Insight.Visibility.VISIBLE }
```

Comment:
A few issues I see with this:

* This will only ever show the first insight that was reported. Subsequent insights will not be shown since you're using `first`
* There's no way to clear an insight once you're displayed it. In our specific case, we want to clear the hierarchy message once the hierarchy call is no longer taking too long.
* Insights are all stored in memory for the entire session. (This might be ok if we don't expect to be storing many insights, but that might not be a safe assumption)

Reply:
Ok updated:

1. Removed the first call, instead we return each and every insight we get from callback also mapping to different class, BannerMessage.
2. For clearing insight I've also added updated callback to clear function of Insights
3. For this one I'll not defer any kind of storage optimization just yet, since insights are less. But agreed if this comes to more insights we can back this with a storage.